packages/headless/src/features/commerce/common/base-commerce-api-request-builder.ts

```diff
@@ -30,7 +30,7 @@ export const buildBaseCommerceAPIRequest = (
         state.configuration.environment
       ),
     organizationId: state.configuration.organizationId,
-    trackingId: state.configuration.analytics.trackingId,
+    trackingId: state.configuration.analytics.trackingId ?? null,
```

Comment:
[Commerce API](https://platformdev.cloud.coveo.com/docs?urls.primaryName=Commerce#/Search/rest_organizations_paramId_commerce_v2_search_post) doesn't accept `null` so falling back to it doesn't reflect the actual contract:

![image](https://github.com/user-attachments/assets/748fd987-f831-4d3d-88b6-3f3f784d2b5b)

Reply:
That is why I did a first pass at it, now I will need someone from commerce to provide a different way to handle the optional `trackingId` then :) 

I tried to give you a solution that would match what the rest of the code would be doing, but as you point out it won't be possible for the commerce engine.

Is it a good idea btw to have that same parameter but its definition differ from the events api?

Reply:
@samisayegh It looks like there is already a validation made when building the commerce engine for the configuration parameters.

https://github.com/coveo/ui-kit/blob/fdf1941adef23209c4b8d36baae75d41a1bfec6d/packages/headless/src/app/commerce-engine/commerce-engine.ts#L189

If that's okay you can rely on that, and I can make sure that trackingId is in fact not null and not undefined in the request builder?

Reply:
Could be this instead, that would validate the typings, and assume that `trackingId` is not `null` or `undefined` at that moment. It could still crash at runtime though if someone manges to make a `commerce-api` request while not using the `commerce-engine` ? Is that possible?

```suggestion
    trackingId: state.configuration.analytics.trackingId!,
```

Reply:
Using `!` is better, but can't the `trackingId` in commerce engine remain just a string?

To use commerce api via headless, one needs to use commerce engine. I wouldn't consider other paths,

Reply:
Yea if I use the `!` instead, then I can keep it as a string in the commerce-engine type if that's how you prefer to handle it.

Reply:
Reverted the changes to commerce-api-request builder